yarn-project/foundation/src/fields/fields.ts

```diff
@@ -158,6 +158,7 @@ abstract class BaseField {
 
 /**
  * Constructs a field from a Buffer of BufferReader.
+ * TODO: should we not throw an error if the size of the input buffer exceeds BaseField.SIZE_IN_BYTES?
```

Comment:
We should not because it's common that we are deserializing a larger object than 1 field from a buffer and in those cases we use the BufferReader flow. Example [here](https://github.com/AztecProtocol/aztec-packages/blob/4908df898f8152a6c95c56d7fce1f935382b60e6/yarn-project/circuits.js/src/structs/tx_context.ts#L76).